### PR TITLE
Fix Error Rendering .ipynb with Array in 'text/plain' Output

### PR DESCRIPTION
**Pull request by itsmevichu**
Fix for - https://github.com/jupyterlab/jupyterlab/issues/16623

### CODE REVIEW

File: packages/notebook/src/windowing.ts [hunk 1]
```diff
@@ -61,9 +61,12 @@ export class NotebookViewModel extends WindowedListModel {
...
       for (let outputIdx = 0; outputIdx < model.outputs.length; outputIdx++) {
         const output = model.outputs.get(outputIdx);
-        outputsLines = ((output.data['text/plain'] as string) ?? '').split(
-          '\n'
-        ).length;
+        const data = output.data['text/plain'];
+        if (typeof data === 'string') {
+          outputsLines += data.split('\n').length;
+        } else if (Array.isArray(data)) {
+          outputsLines += data.join('\n').split('\n').length;
```
Comment (krassowski) [inline packages/notebook/src/windowing.ts:68]: I need to double check this but I think the new lines should be included in the array.

```suggestion
          outputsLines += data.join('').split('\n').length;
```

File: packages/notebook/test/modelfactory.spec.ts [hunk 2]
```diff
@@ -79,4 +85,72 @@ describe('@jupyterlab/notebook', () => {
...
+      const height = notebook.estimateWidgetSize(0);
+      expect(height).toBe(56); // (1 source_line + 1 output_line) * 17 (line_height) + 22 (cell_margin)
+    });
+
+    test('should calculate height based on number of lines in source and output (string[] output)', () => {
+      const outputObj = [
+        {
+          output_type: 'execute_result',
+          data: {
+            'text/plain': ['Output line 1', 'Output line 2', 'Output line 3'],
```
Comment (krassowski) [inline packages/notebook/test/modelfactory.spec.ts:129]:
```suggestion
            'text/plain': ['Output line 1\n', 'Output line 2\n', 'Output line 3\n'],
```

image

File: packages/notebook/test/modelfactory.spec.ts [hunk 3]
```diff
@@ -79,4 +85,72 @@ describe('@jupyterlab/notebook', () => {
...
+          },
+          execution_count: 4,
+          metadata: {}
+        }
+      ];
+      const sharedModel = createStandaloneCell({
+        cell_type: 'code',
+        execution_count: 1,
+        outputs: outputObj,
+        source: ['for i in range(3):\n', '    print(f"Line {i+1}")'],
```
Comment (krassowski) [inline packages/notebook/test/modelfactory.spec.ts:140]: Just to align with output:

```suggestion
        source: ['for i in range(3):\n', '    print(f"Output line {i+1}")'],
```

File: packages/notebook/test/modelfactory.spec.ts (hunk 2 above)
Comment (krassowski) [reply packages/notebook/test/modelfactory.spec.ts:129]: This is the format that I see in ipykernel and IRkernel

**Review by krassowski — COMMENTED** [review]
It does look like the array elements already contain `\n` to me

File: packages/notebook/test/modelfactory.spec.ts (hunk 2 above)
Comment (itsmevichu) [reply packages/notebook/test/modelfactory.spec.ts:129]: I agree with you, sorry for the misinterpretation. Will adapt the change.

**Review by krassowski — APPROVED** [review]
LGTM, thanks @itsmevichu!

**Pull request merged**